torch/csrc/distributed/autograd/functions/recvrpc_backward.h

```diff
@@ -20,7 +21,7 @@ class TORCH_API RecvRpcBackward : public torch::autograd::Node {
  public:
   explicit RecvRpcBackward(
       const AutogradMetadata& autogradMetadata,
-      DistAutogradContext& autogradContext,
+      std::shared_ptr<DistAutogradContext> autogradContext,
```

Comment:
Why this is not using `ContextPtr`?

Reply:
Would have a circular dependency, thats why we need to have a forward declaration for `DistAutogradContext`.